be/src/runtime/routine_load/data_consumer.cpp

```diff
@@ -384,13 +384,21 @@ Status KafkaDataConsumer::get_offsets_for_times(const std::vector<PIntegerPair>&
 
 // get latest offsets for given partitions
 Status KafkaDataConsumer::get_latest_offsets_for_partitions(
-        const std::vector<int32_t>& partition_ids, std::vector<PIntegerPair>* offsets) {
+        const std::vector<int32_t>& partition_ids, std::vector<PIntegerPair>* offsets,
+        int timeout) {
+    MonotonicStopWatch watch;
+    watch.start();
     for (int32_t partition_id : partition_ids) {
         int64_t low = 0;
         int64_t high = 0;
+        auto timeout_ms = timeout - watch.elapsed_time() / 1000 / 1000;
```

Comment:
watch.elapsed_time() return uint64_t, implicit type conversion may lead to unexpected issues.